src/content/docs/reference/Uniforms/Matrices/textureMatrix.mdx

```diff
@@ -5,11 +5,13 @@ sidebar:
   label: textureMatrix
   order: 1
   badge:
-    text: Unfinished
-    variant: caution
+    text: 1.17+
+    variant: danger
 ---
 import { Aside } from '@astrojs/starlight/components';
 
 <Aside type="danger">This uniform only works on `1.17` and newer.</Aside>
 
-### `uniform ;`
\ No newline at end of file
+### `uniform mat4 textureMatrix;`
+
+Always equal to `mat4(1.0)`.
```

Comment:
This is not true, enchantment glint uses the texture matrix to scroll the texture (and maybe also the end portal texture you'd have to investigate that)

Reply:
That would make more sense, but in every context I've tested, it's always equal to `mat4(1.0)`, which is also stated in [OptiDocs](https://optifine.readthedocs.io/shaders_dev/uniforms.html?h=texturematrix#general-uniforms).

Do you know of any way to test the value of `textureMatrix` that shows it's different?

Reply:
Yeah, the Optifine docs are not very complete, they also state the `mat4(1.0)` is the default, not that it's always that value. Since it's a uniform, you should be able to write it to the screen in the gbuffers_armor_glint.fsh program (which handles enchantment glint). Also seems like the end portal does not change the texture matrix.

Reply:
Turns out I accidentally used the wrong program when checking the enchantment glint 😆
Using it in `gbuffers_armor_glint` works like expected👍